Main.hs

```diff
@@ -1,107 +1,95 @@
+{-# LANGUAGE MultiWayIf, BangPatterns #-}
+
 module Main where
 
 import Control.Monad
 import System.IO
+import qualified Data.List as List
 
-square :: Int -> Pic
-square n = Pic $ replicate n (replicate n '#')
+newtype Tixel = Tixel { unTixel :: Char }
 
-triangle :: Int -> Pic
-triangle = Pic . aux
-  where
-    aux :: Int -> [String]
-    aux 0 = []
-    aux n = replicate n 'T':aux (n-1)
+instance Semigroup Tixel where
+  tx1@(Tixel px1) <> tx2 = if px1 == ' ' then tx2 else tx1
 
-circle :: Int -> Pic
-circle r = Pic [ [ if x*x + y*y < r*r then 'S' else ' '
-                 | x <- [-r..r]
-                 ]
-               | y <- [-r..r]
-               ]
+instance Monoid Tixel where
+  mempty = Tixel ' '
 
-newtype Pic = Pic [String]
+newtype Pic a = Pic ((Int, Int) -> a)
```

Comment:
What does this represent conceptually? A Pic is a mapping from the Cartesian plane to some `a`?

Reply:
Yep! I've seen all the stuff you had to do to be able to transform and compose them so I thought this might be a bit easier.
TBH this was strongly inspired by an example we were shown in a Haskell course at Uni.
The only reason I gave it a type parameter is because you had a comment regarding how to use it as a functor which got me thinking.